fluence-app-service/src/service.rs

```diff
@@ -130,19 +102,19 @@ impl AppService {
         let local_dir: String = local_dir_path.to_string_lossy().into();
         let tmp_dir: String = tmp_dir_path.to_string_lossy().into();
 
-        let service_id_env = vec![format!("{}={}", SERVICE_ID_ENV_NAME, service_id).into_bytes()];
         let preopened_files = vec![local_dir.clone(), tmp_dir.clone()];
         let mapped_dirs = vec![
             (String::from(SERVICE_LOCAL_DIR_NAME), local_dir),
             (String::from(SERVICE_TMP_DIR_NAME), tmp_dir),
         ];
+        envs.push(format!("{}={}", SERVICE_ID_ENV_NAME, service_id));
 
         config.modules_config = config
             .modules_config
             .into_iter()
             .map(|(name, module_config)| {
                 let module_config = module_config
-                    .extend_wasi_envs(service_id_env.clone())
+                    .extend_wasi_envs(envs.iter().map(|s| s.clone().into_bytes()).collect())
```

Comment:
Seems that extend_wasi_envs should receive `impl Iterator<item = Vec<u8>>`, if you don't mind I will rewrite this myself later.